.github/workflows/release_staging.yml

```diff
@@ -37,21 +42,47 @@ jobs:
     - name: Versioning
       uses: "./.github/actions/versioning"
       with:
-        release_type: staging
+        RELEASE_TYPE: ${{ env.RELEASE_TYPE }}
+    - name: Extract version
+      id: extract_version
+      run: echo "RELEASE_VERSION=${version}" >> $GITHUB_OUTPUT
     - name: Publish to Cloudflare Pages Staging 
       uses: "./.github/actions/publish_to_pages_staging"
       with:       
         CLOUDFLARE_ACCOUNT_ID: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
         CLOUDFLARE_API_TOKEN: ${{ secrets.CLOUDFLARE_API_TOKEN }}
-
-    - name: Build Docker image and push to Docker hub and K8S
-      uses: "./.github/actions/build_and_push_docker_image"
+    - name: Upload Build Artifact
+      uses: actions/upload-artifact@v4
       with:
-        DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
-        DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
-        DOCKERHUB_ORGANISATION: ${{ secrets.DOCKERHUB_ORGANISATION }}
-        K8S_NAMESPACE: ${{ secrets.K8S_NAMESPACE }}
-        KUBE_SERVER: ${{ secrets.KUBE_SERVER }}
-        SERVICEACCOUNT_TOKEN: ${{ secrets.SERVICEACCOUNT_TOKEN }}
-        CA_CRT: ${{ secrets.CA_CRT }}
-        APP_VERSION: latest-staging
+        name: build
+        path: .
+        retention-days: 1
+
+  build_and_publish_to_docker_k8s:
```

Comment:
no slack notification is needed for staging workflow. please revert the changes here.